src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+        mov(reg_buffer_aux, reg_buffer);
+        L(sub_exp_reduce_loop_label);
+        {
+            cmp(reg_work_amount_aux, vec_size);
+            jl(sub_exp_reduce_end_label, T_NEAR);
+
+            sub_exp_reduce(vec_size);
+
+            sub(reg_work_amount_aux, vec_size);
+
+            jmp(sub_exp_reduce_loop_label, T_NEAR);
+        }
+        L(sub_exp_reduce_end_label);
+        if (tail_size) {
+            sub_exp_reduce(tail_size);
+        }
+
+        sub(rsp, sizeof(float) * vec_size);
+        uni_vmovups(ptr[rsp], get_vmm_denom(0));
+        uni_vpxor(get_vmm_aux(0), get_vmm_aux(0), get_vmm_aux(0));
+        for (size_t i = 0; i < vec_size; i++) {
+            mov(reg_tmp_32, ptr[rsp + i * sizeof(float)]);
+            vmovq(xmm_tmp, reg_tmp);
+            uni_vaddps(get_xmm_aux(0), get_xmm_aux(0), xmm_tmp);
+        }
+        vbroadcastss(get_vmm_aux(0), get_xmm_aux(0));
+        add(rsp, sizeof(float) * vec_size);
+
+        mov(reg_tmp, dnnl::impl::float2int(1.0f));
+        vmovq(xmm_tmp, reg_tmp);
+        vbroadcastss(get_vmm_denom(0), xmm_tmp);
+        uni_vdivps(get_vmm_denom(0), get_vmm_denom(0), get_vmm_aux(0));
+
+        if (jcp_.with_scales1)
+            mov(reg_scales, ptr[reg_params + GET_OFF(p_scales1)]);
+
+        if (jcp_.with_scales1 && jcp_.broadcast_scales1) {
+            uni_vmovss(Xmm(vmm_scales.getIdx()), ptr[reg_scales]);
+            uni_vbroadcastss(vmm_scales, Xmm(vmm_scales.getIdx()));
+        }
+
+        mov(reg_work_amount_aux, reg_work_amount);
+        L(mul_loop_label);
+        {
+            cmp(reg_work_amount_aux, vec_size);
+            jl(mul_end_label, T_NEAR);
+
+            mul_loop(vec_size);
+
+            sub(reg_work_amount_aux, vec_size);
+
+            jmp(mul_loop_label, T_NEAR);
+        }
+        L(mul_end_label);
+        if (tail_size) {
+            mul_loop(tail_size);
+        }
+
+        this->postamble();
+
+        load_emitter->emit_data();
+        store_emitter->emit_data();
+        exp_emitter->emit_data();
+    }
+
+    void mul_add_max(size_t step) {
+        bool is_tail = step < vec_size;
+
+        load(get_vmm_in(0), reg_mul_in0, jcp_.src_prc, step, is_tail);
+        load(get_vmm_in(2), reg_add_in1, Precision::FP32, step, is_tail);
+
+        if (jcp_.src_prc == Precision::I32) {
+            if (jcp_.with_scales0) {
+                if (!jcp_.broadcast_scales0) {
+                    load(vmm_scales, reg_scales, Precision::FP32, step, is_tail);
+                    add(reg_scales,  sizeof(float) * step);
+                }
+                uni_vmulps(get_vmm_in(0), get_vmm_in(0), vmm_scales);
+                uni_vmaxps(get_vmm_in(0), get_vmm_in(0), vmm_crop_low);
+                uni_vminps(get_vmm_in(0), get_vmm_in(0), vmm_crop_high);
+            }
+
+            uni_vaddps(get_vmm_aux(0), get_vmm_in(0), get_vmm_in(2));
+            uni_vmulps(get_vmm_aux(0), get_vmm_aux(0), get_vmm_in(1));
```

Comment:
```suggestion
            uni_vmulps(get_vmm_aux(0), get_vmm_aux(0), get_vmm_in(1));
            uni_vaddps(get_vmm_aux(0), get_vmm_in(0), get_vmm_in(2));
```
is it intentional? otherwise the order matters.

Reply:
Actual order of operations depends on type of MHA pattern. Reworked this part: added explicit flag `mulBeforeAdd`.